src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -217,6 +227,40 @@ public void start(final Map<String, String> props) {
     return null;
   }
 
+  private Optional<String> convertToJson(Optional<BsonDocument> jsonDocumentraw, JsonWriterSettings jsonOptions) {
+        if(jsonOptions!=null && jsonDocumentraw.isPresent()){
+          return Optional.of(jsonDocumentraw.get().toJson(jsonOptions));
+        }
+        else if(jsonDocumentraw.isPresent()){
+          return Optional.of(jsonDocumentraw.get().toJson());
+        }
+        else{
+          return Optional.empty();
+        }
```

Comment:
Here, you can use [`Optional<T>.map()`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#map-java.util.function.Function-) to improve readability:
```suggestion
    if (jsonWriterSettings == null) {
      return document.map(BsonDocument::toJson);
    }

    return document.map((doc) -> doc.toJson(jsonWriterSettings));
```